quickevent/app/quickevent/plugins/CardReader/src/services/racomclient.cpp

```diff
@@ -132,6 +135,85 @@ void RacomClientSirxdConnection::onReadyRead()
 	}
 }
 
+RacomReadSplitFile::RacomReadSplitFile(QString fileName, int interval, int finishCode, QObject *parent)
+	: Super(parent)
+	, m_fileName(fileName)
+	, m_interval(interval)
+	, m_finishCode(finishCode)
+{
+	m_ReadTimer = new QTimer(this);
+	connect(m_ReadTimer, &QTimer::timeout, this, &RacomReadSplitFile::onTimerTimeOut);
+}
+
+RacomReadSplitFile::~RacomReadSplitFile()
+{
+	m_ReadTimer->stop();
+	QF_SAFE_DELETE(m_ReadTimer)
+}
+
+void RacomReadSplitFile::run()
+{
+	m_ReadTimer->start(m_interval*1000);
+}
+
+void RacomReadSplitFile::readAndProcessFile()
+{
+	QStringList text;
+	try {
+			QFile file(m_fileName);
+			if(file.open(QIODevice::ReadOnly|QIODevice::Text))
+			{
+				QTextStream stream(&file);
+
+				QString line;
+				do {
+					line = stream.readLine();
+					text << line;
+				} while(!line.isNull());
+			}
+			file.close();
+	}  catch (QException &e) {
```

Comment:
which function in try/catch can throw QException?

Reply:
Not sure now which function exactly.
It is there for catch if file cannot be open or read - racom file is on network device (OOk server)

Reply:
QException is not thrown in this case. https://doc.qt.io/qt-5/qexception.html#details

Reply:
I'll try to check it and fix it.

Reply:
any progress?

Reply:
Yes, I've remove try/catch